src/main/java/com/ververica/flink/table/gateway/operation/InsertOperation.java

```diff
@@ -186,6 +186,9 @@ protected void cancelJobInternal() {
 		// blocking deployment
 		try {
 			JobClient jobClient = deployer.deploy().get();
+			if (isYarnPerJobMode){
```

Comment:
Is it possible to extract these yarn per job logic into a new class? We might add k8s support in the future and it is very likely that k8s per job mode also needs a series of specific logic.

Reply:
Got it. I will refactor these codes. Thanks for your suggestions.